updater/developer_mode.php

```diff
@@ -0,0 +1,144 @@
+	//allow to access this page
+}
+else {
+	die("You are not allowed to handle updater administration");
+}
+if ($updater_host == "github") {
+	//if host=github then load
+	if (getUpdaterSetting("github_current") != "empty_setting") {
+		$pull_request_number = getUpdaterSetting("github_current");
+	}
+	else {
+		//load settings from json file, this pr number refers to the pr number at which the updater gets merged
+		//replace # since it is not fit for api
+		$pull_request_number = $settings_array['github_current'];
+		$pull_request_number = str_replace("#", "", $pull_request_number);
+	}
+//LOADING API FUNCTIONS ACCORDING TO THE HOST VALUE
+
+}
+
+	require "lib/api.$updater_host.php";
+if (getUpdaterSetting("updater_dm") == "empty_setting" || getUpdaterSetting("updater_dm")== "0") {
+	echo "<p style='color: red;'>Developer Mode is not Enabled</p>";
+}
+elseif (getUpdaterSetting("updater_token") == "empty_setting") {
+	echo "<p style='color: red;'>You didnt add the token to updater yet, first setup the updater and access developer mode</p>";
+
+}
+elseif (!isTokenValid(getUpdaterSetting("updater_token"))) {
+		echo "<p style='color: red;'>The token was expired, run the updater to add the new token</p>";
+}
+elseif (getUpdaterSetting("updater_dm") == "1") {
+	if (getUpdaterSetting("github_developer_current") == "empty_setting"){
+
+	}
+	else {
+		$pr = getUpdaterSetting("github_developer_current");
+		echo "<div class='alert alert-info'><table class='table table-striped'><tr><td>#$pr has already applied.</td><td><button id='restore_button' class='css-button cp-submit'><i class='fa fa-refresh'></i> Restore</button></td></div>";
+		// show a backup and sync button so the user can go back to state before PR applied
+	}
+	//Means Developer Mode is Enabled for Updater
+	//Show a list of pull requests which are open
+	$pr_array = getAllOpenPullRequests($access_token, $repository_owner, $repository_name);
+	echo "<table class='table table-striped' id='pr_interface'>";
+	foreach ($pr_array as $key) {
+		$avatar_url = $key['user']['avatar_url'];
+		$pr_title = $key['title'];
+		$pr_number = $key['number'];
+		$pr_body = substr($key['body'], 0,80);
+		$user_name = $key['user']['login'];
+		$loader->set_template_file("developer_mode_pr");
+		$loader->assign("AVATAR_URL", $avatar_url);
+		$loader->assign("TITLE", $pr_title);
+		$loader->assign("BODY", $pr_body);
+		$loader->assign("USERNAME", $user_name);
+		$loader->assign("PR_NUMBER", $pr_number);
+		$loader->output();
+	}
+	echo "</table>";
+	$loader->set_template_file("developer_mode_progress");
+	$loader->assign("AVATAR_URL", $avatar_url);
+	$loader->output();
+}
+
+
+?>
+<script type="text/javascript">
+	    parent.$('.iziModal-content').css("background-image", "none");
+	        $('.test').click(function() {
+	        	var id = $(this).attr('id');
+    	var bool = prompt("Previous Testing of the pr will be returned to original state.enter confirm to test the pull request.");
```

Comment:
... translation required ... really on all fixed text strings, will not repeat on every occurrence but assume I did :-)